hyph-utf8/source/generic/hyph-utf8/languages/mul-ethi/generate_patterns_mul-ethi.lua

```diff
@@ -49,26 +89,48 @@ local field = (1 - colon)^0
 local eth_syll = P"ETHIOPIC SYLLABLE " * C(field)
 local ucd_entry = C(digit^4) * colon * eth_syll
 
-local pattfile = assert(io.open("hyph-mul-ethi.tex", "w"))
+local pattfile = assert(io.open("../../../../../tex/generic/hyph-utf8/patterns/tex/hyph-mul-ethi.tex", "w"))
+
+
+extract_mit_licence()
 
 pattfile:write[[
-% Experimental pattern file for languages written using the Ethiopic script.
-% Arthur Reutenauer, London, 2011, for the hyph-utf8 project.
-% Copyright (c) TeX Users Group, 2011.
-% You may freely use, copy, modify and / or redistribute this file.
-%
+% title: Experimental pattern file for languages written using the Ethiopic script.
+% copyright: Copyright (c) 2011, 2016 Arthur Reutenauer
+% notice: This file is part of the hyph-utf8 package.
+%     See http://www.hyphenation.org for more information
+% language:
+%     name: Multiple languages using the Ethiopic scripts
+%     tag: mul-ethi
+% version: 0.2 2016-05-22
+% licence:
+%     - name: MIT
+%     - url: https://opensource.org/licenses/MIT
+%     - text: >
+]]
+
+for _, line in ipairs(mit_licence) do
+  if lpeg.match(lpeg.P(-1), line) then
+    pattfile:write('%\n')
+  else
+    pattfile:write('%         ', line, '\n')
+  end
+end
+
+pattfile:write[[
+% ==========================================
 % This is a generated file.  If you wish to edit it, consider adapting the
 % generating programme
-% (svn://tug.org/texhyphen/trunk/hyph-utf8/source/generic/hyph-utf8/languages/mul-ethi/generate_patterns_mul-ethi.lua).
+% (https://github.com/hyphenation/tex-hyphen/blob/master/hyph-utf8/source/generic/hyph-utf8/languages/mul-ethi/generate_patterns_mul-ethi.lua).
 %
-% The BCP 47 language tag for that file is "mul-ethi" to reflect the fact that
+% The BCP 47 language tag for that file is “mul-ethi” to reflect the fact that
```

Comment:
I would prefer to keep the plain quotation marks.

Reply:
We’ll keep the Unicode ones, they look nicer.